playbooks/adhoc/upgrades/upgrade.yml

```diff
@@ -156,7 +156,7 @@
         to_version: '3.1'
         role: master
         config_base: "{{ hostvars[inventory_hostname].openshift.common.config_base }}"
-      when: deployment_type in ['openshift-enterprise', 'atomic-enterprise'] and g_aos_versions.curr_version | version_compare('3.1', '>=')
+      when: deployment_type in ['openshift-enterprise', 'atomic-enterprise'] and hostvars[groups.masters.0].g_aos_versions.curr_version | version_compare('3.1', '>=')
```

Comment:
g_new_version is also tied to the first master but it should probably be used instead